angel_system/activity_classification/tcn_hpl/predict.py

```diff
@@ -464,19 +464,29 @@ def collect(
                 raise RuntimeError(
                     "No video set before results collection. See `set_video` method."
                 )
-            packet = dict(
+            # get the global id for the image from the frame number
+
+            # add the image
+            img = dict(
                 video_id=self._vid,
                 frame_index=frame_index,
-                activity_pred=activity_pred,
-                activity_conf=list(activity_conf_vec),
             )
             if name is not None:
-                packet["name"] = name
+                img["name"] = name
             if file_name is not None:
-                packet["file_name"] = file_name
+                img["file_name"] = file_name
             if activity_gt is not None:
-                packet["activity_gt"] = activity_gt
-            self._dset.add_image(**packet)
+                img["activity_gt"] = activity_gt
```

Comment:
I don't think we should still be adding this non-standard field to the image objects. As long as we ensure there are equivalent image entries and that the added annotations reference their relevant images, the output COCO file will be separately interpretable to the ground truth file that was input.

Reply:
agreed - I'll drop this.